cmd/hermit/main.go

```diff
@@ -0,0 +1,293 @@
+package main
+
+import (
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"math"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/klarna/eremetic"
+	"github.com/klarna/eremetic/client"
+)
+
+var defaultEremeticServer = "http://localhost:8000"
+
+func main() {
+	runCmd := newSubCommand("run",
+		"hermit run [OPTION]... COMMAND",
+		"Run an eremetic task.")
+	runOpts := new(runOptions)
+	runOpts.Register(runCmd)
+
+	taskCmd := newSubCommand("task",
+		"hermit task TASK",
+		"Fetch information about a specific task.")
+
+	listCmd := newSubCommand("ls",
+		"hermit ls [OPTION]...",
+		"List active eremetic tasks.")
+	listOpts := new(listOptions)
+	listOpts.Register(listCmd)
+
+	logsCmd := newSubCommand("logs",
+		"hermit logs [OPTION]... TASK",
+		"Fetch the logs of a task.")
+	logsOpts := new(logsOptions)
+	logsOpts.Register(logsCmd)
+
+	versionCmd := newSubCommand("version",
+		"hermit version",
+		"Show the Eremetic version information.")
+
+	if len(os.Args) < 2 {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	switch os.Args[1] {
+	case "run":
+		runCmd.Parse(os.Args[2:])
+	case "task":
+		taskCmd.Parse(os.Args[2:])
+	case "ls":
+		listCmd.Parse(os.Args[2:])
+	case "logs":
+		logsCmd.Parse(os.Args[2:])
+	case "version":
+		versionCmd.Parse(os.Args[2:])
+	default:
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	srv := os.Getenv("EREMETIC_URL")
+	if srv == "" {
+		srv = defaultEremeticServer
+	}
+
+	httpClient := &http.Client{}
+
+	if os.Getenv("HERMIT_INSECURE") != "" {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		httpClient.Transport = tr
+	}
+
+	ec, err := client.New(srv, httpClient)
+	if err != nil {
+		exitWithError(err)
+	}
+
+	if runCmd.Parsed() {
```

Comment:
Could these be broken into their own functions to make the main func a bit shorter and easier to read?